src/ptvsd/_vendored/pydevd/tests_python/test_debugger_json.py

```diff
@@ -2159,6 +2161,135 @@ def test_source_mapping(case_setup):
+                return True
+            return 'ENGINE ' in line or 'CherryPy Checker' in line or 'has an empty config' in line
+
+        writer._ignore_stderr_line = _ignore_stderr_line
+
+        json_facade = JsonFacade(writer)
+        json_facade.write_launch(debugOptions=['DebugStdLib'])
+
+        break1_line = writer.get_line_index_with_content('break here')
+        json_facade.write_set_breakpoints(break1_line)
+
+        server_socket = writer.server_socket
+
+        class SecondaryProcessWriterThread(AbstractWriterThread):
+
+            TEST_FILE = writer.get_main_filename()
+            _sequence = -1
+
+        class SecondaryProcessThreadCommunication(threading.Thread):
+
+            def run(self):
+                from tests_python.debugger_unittest import ReaderThread
+                expected_connections = 1
+                for _ in range(expected_connections):
+                    server_socket.listen(1)
+                    self.server_socket = server_socket
+                    new_sock, addr = server_socket.accept()
+
+                    reader_thread = ReaderThread(new_sock)
+                    reader_thread.name = '  *** Multiprocess Reader Thread'
+                    reader_thread.start()
+
+                    writer2 = SecondaryProcessWriterThread()
+
+                    writer2.reader_thread = reader_thread
+                    writer2.sock = new_sock
+
+                    writer2.write_version()
+                    writer2.write_add_breakpoint(break1_line)
+                    writer2.write_make_initial_run()
+
+                # Give it some time to startup
+                time.sleep(2)
+                t = writer.create_request_thread('http://127.0.0.1:%s/' % (port,))
+                t.start()
+
+                hit = writer2.wait_for_breakpoint_hit()
+                writer2.write_run_thread(hit.thread_id)
+
+                contents = t.wait_for_contents()
+                assert 'Hello World NEW!' in contents
+
+                t = writer.create_request_thread('http://127.0.0.1:%s/exit' % (port,))
+                t.start()
+
+        secondary_process_thread_communication = SecondaryProcessThreadCommunication()
+        secondary_process_thread_communication.start()
+        json_facade.write_make_initial_run()
+
+        # Give it some time to startup
+        time.sleep(2)
+
+        t = writer.create_request_thread('http://127.0.0.1:%s/' % (port,))
+        t.start()
+        json_facade.wait_for_thread_stopped()
+        json_facade.write_continue()
+
+        contents = t.wait_for_contents()
+        assert 'Hello World INITIAL!' in contents
+
+        # Sleep a bit more to make sure that the initial timestamp was gotten in the
+        # CherryPy background thread.
+        time.sleep(2)
+        f.write(tmplt % dict(port=port, str='NEW'))
+
+        secondary_process_thread_communication.join(10)
+        if secondary_process_thread_communication.isAlive():
```

Comment:
camelcased version of this method is deprecated
```suggestion
        if secondary_process_thread_communication.is_alive():
```